$MirOS: ports/emulators/qemu/snapshot/patches/patch-fpu_softfloat-native_c,v 1.1 2008/05/07 15:53:10 tg Exp $
$OpenBSD: patch-fpu_softfloat-native_c,v 1.2 2007/05/01 12:55:14 todd Exp $
--- fpu/softfloat-native.c.orig	Wed Sep 23 19:01:08 2009
+++ fpu/softfloat-native.c	Sat Oct 17 16:52:46 2009
@@ -1,11 +1,16 @@
 /* Native implementation of soft float functions. Only a single status
    context is supported */
 #include "softfloat.h"
+#include "gnu-c99-math.h"
 #include <math.h>
 #if defined(HOST_SOLARIS)
 #include <fenv.h>
 #endif
 
+#ifdef HOST_BSD
+#include <sys/param.h>
+#endif
+
 void set_float_rounding_mode(int val STATUS_PARAM)
 {
     STATUS(float_rounding_mode) = val;
@@ -26,7 +31,8 @@ void set_floatx80_rounding_precision(int
 }
 #endif
 
-#if defined(HOST_BSD) || (defined(HOST_SOLARIS) && HOST_SOLARIS < 10)
+#if (defined(HOST_BSD) && (!defined(MirBSD) || (MirBSD < 0x09A6))) || \
+    (defined(HOST_SOLARIS) && HOST_SOLARIS < 10)
 #define lrint(d)		((int32_t)rint(d))
 #define llrint(d)		((int64_t)rint(d))
 #define lrintf(f)		((int32_t)rint(f))
